Remove count_view_mes

valadaptive opened the pull request:
As far as I can tell, the only time it goes out of sync with chat.length is in printMessages. For that use case, I just pass in the intended message ID to `forceID`, and I've allowed a message ID to be passed to `hideSwipeButtons` and `showSwipeButtons` so they can act on the proper message as well.

valadaptive committed:
Remove count_view_mes

Cohee1207 commented:
Something is wrong here. Sending a regular message with streaming causes it to stream into the user message, replacing its contents.

![image](https://github.com/SillyTavern/SillyTavern/assets/18619528/970c2ee1-8281-4a60-98ea-85a39ec80e24)

valadaptive committed:
Fix off-by-one in addOneMessage

valadaptive committed:
Remove getMessageId

As far as I can tell, we don't add/remove anything from chat in between
the calculation of newMessageId and subsequent calls to getMessageId.
We can just use newMessageId everywhere.

valadaptive commented:
There was a silly off-by-one in there. Try again?

Cohee1207 commented on public/script.js:
```diff
@@ -1904,7 +1900,7 @@ function addOneMessage(mes, { type = 'normal', insertAfter = null, scroll = true
     }
 
-    if (count_view_mes == 0) {
+    if (chat.length == 0) {
```
This feels like a dead code now. If chat length is 0, then addOneMessage won't ever trigger

valadaptive replied:
Good catch; I've removed it

valadaptive committed:
Remove dead if statement

Cohee1207 commented:
Seems to be working now for all of the cases I could think of.
